src/Google/GlobalSiteTag.php

```diff
@@ -155,13 +189,164 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
+				'{
+				id: "gla_%s",
+				price: %s,
+				google_business_vertical: "retail",
+				name:"%s",
+				quantity:"%s",
+				}',
+				esc_js( $product_id ),
+				esc_js( $price ),
+				esc_js( $product_name ),
+				esc_js( $quantity ),
+			);
+		}
+
+		// Check if this is the first time customer
+		$is_new_customer = $this->is_first_time_customer( $order->get_billing_email() );
+
+		// Track the purchase page
+		$language = $this->wp->get_locale();
+		if ( 'en_US' === $language ) {
+			$language = 'English';
+		}
+		$purchase_page_gtag =
+		sprintf(
+			'gtag("event", "purchase", {
+			ecomm_pagetype: "purchase",
+			send_to: "%s",
+			transaction_id: "%s",
+			currency: "%s",
+			country: "%s",
+			value: "%s",
+			new_customer: %s,
+			tax: "%s",
+			shipping: "%s",
+			delivery_postal_code: "%s",
+			aw_feed_country: "%s",
+			aw_feed_language: "%s",
+			items: [%s]});',
+			esc_js( "{$ads_conversion_id}/{$ads_conversion_label}" ),
+			esc_js( $order->get_id() ),
+			esc_js( $order->get_currency() ),
+			esc_js( $this->wc->get_base_country() ),
+			esc_js( $order->get_total() ),
+			esc_js( $is_new_customer ),
+			esc_js( $order->get_cart_tax() ),
+			esc_js( $order->get_total_shipping() ),
+			esc_js( $order->get_billing_postcode() ),
+			esc_js( $order->get_shipping_postcode() ),
+			esc_js( $this->wc->get_base_country() ),
+			esc_js( $language ),
+			join( ',', $item_info ),
+		);
+		wp_print_inline_script_tag( $purchase_page_gtag );
+	}
+
+	/**
+	 * Display the JavaScript code to track the product view page.
+	 */
+	private function display_view_item_event_snippet(): void {
+		// Only display on the product view page.
+		if ( ! is_product() ) {
+			return;
+		}
+		$product        = wc_get_product( get_the_ID() );
+		$view_item_gtag = sprintf(
+			'gtag("event", "view_item", {
+			send_to: "GLA",
+			ecomm_pagetype: "product",
+			value: %s,
+			items:[{
+				id: "gla_%s",
+				price: %s,
+				google_business_vertical: "retail",
+				name: "%s",
+				category: "%s",
+			}]});',
+			esc_js( (float) $product->get_price() ),
+			esc_js( $product->get_id() ),
+			esc_js( (float) $product->get_price() ),
+			esc_js( $product->get_name() ),
+			esc_js( join( ' & ', $this->product_helper->get_categories( $product ) ) ),
+		);
+		wp_print_inline_script_tag( $view_item_gtag );
+	}
+
+	/**
+	 * Display the JavaScript code to track all pages.
+	 */
+	private function display_page_view_event_snippet(): void {
+		$page_view_gtag = sprintf(
+			'gtag("event", "page_view", {
+			send_to: "GLA",});',
+		);
+		if ( ! is_cart() ) {
+			wp_print_inline_script_tag( $page_view_gtag );
+		} else {
+			// display the JavaScript code to track the cart page
+			$item_info = [];
+
+			foreach ( WC()->cart->get_cart() as $cart_item ) {
+				// gets the product id
+				$id = $cart_item['product_id'];
+
+				// gets the product object
+				$product = $cart_item['data'];
+				$name    = $product->get_name();
+				$price   = $product->get_price();
```

Comment:
This should also be the price depending on how it's displayed on cart pages.
```suggestion
				$price   = WC()->cart->display_prices_including_tax() ? wc_get_price_including_tax( $product ) : wc_get_price_excluding_tax( $product );
```